python_ta/cfg/graph.py

```diff
@@ -31,9 +31,12 @@ def link_or_merge(self, source: CFGBlock, target: CFGBlock) -> None:
         An "empty" node for this purpose is when source has no statements.
         """
         if source.statements == []:
-            for edge in source.predecessors:
-                edge.target = target
-                target.predecessors.append(edge)
+            if source is self.start:
+                self.start = target
```

Comment:
self.block_count = 1 included here?

Reply:
Good catch, the issue with including `self.block_count = 1` is that we might have multiple instances of `CFGBlock` with the same id. This also applies if we include `self.block_count -= 1` instead (why I think would be more appropriate). A possible fix is that we use the built-in `id()` function to set the id of the blocks. Then we would be able to adjust the `block_count` attribute in the `link_or_merge` method.

Reply:
Yes, this is a good question. Right now `block_count` is used as an "autoincrement" to ensure the block ids are unique, and we'll take the policy that this number never decreases. I would be open to suggestions for a different name for the attribute, or adding a comment explaining its purpose.

Reply:
I see. I thought `block_count` meant the total number of blocks in the graph. I will make a PR with the comment explaining its purpose for now.